### PR TITLE
Fixes, feature, and doc updates

### PR DESCRIPTION
**Pull request by NathanielRN**
# Description

Some easy updates to the downstream repo to clean up this repo. Fixes some small bugs, updates API calls/docs, and adds small features (like an environment variable which controls Sample App log levels).

Another goal is to merge a PR so that we can run tests against the latest upstream release [opentelemetry v1.5.0 & v0.24b0](https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.5.0) and fix any issues if there are any.

### CODE REVIEW

File: integration-test-apps/auto-instrumentation/flask/application.py [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
+
+# trace.set_tracer_provider(
+#     TracerProvider(
+#         active_span_processor=span_processor,
+#         id_generator=AwsXRayIdGenerator(),
+#         # resource=get_aggregated_resources(
+#         #     [
+#         #         AwsEc2ResourceDetector(),
+#         #     ]
+#         # ),
```
Comment (willarmiros) [inline integration-test-apps/auto-instrumentation/flask/application.py:29]: Probably don't need the double comment here?

File: integration-test-apps/manual-instrumentation/flask/create_flask_app.py [hunk 2]
```diff
@@ -7,16 +7,21 @@
...
+import logging
 from opentelemetry import trace
 from opentelemetry.sdk.extension.aws.trace.propagation.aws_xray_format import (
     TRACE_ID_DELIMITER,
     TRACE_ID_FIRST_PART_LENGTH,
     TRACE_ID_VERSION,
 )
 
+if os.environ.get('SAMPLE_APP_LOG_LEVEL') == 'ERROR':
+    log = logging.getLogger('werkzeug')
```
Comment (willarmiros) [inline integration-test-apps/manual-instrumentation/flask/create_flask_app.py:20]: What is `werkzeug`?

File: integration-test-apps/auto-instrumentation/flask/create_flask_app.py [hunk 3]
```diff
@@ -7,16 +7,21 @@
...
     TRACE_ID_DELIMITER,
     TRACE_ID_FIRST_PART_LENGTH,
     TRACE_ID_VERSION,
 )
 
+if os.environ.get('SAMPLE_APP_LOG_LEVEL') == 'ERROR':
+    log = logging.getLogger('werkzeug')
+    log.setLevel(logging.ERROR)
+
 # NOTE: (NathanielRN) Metrics is on hold until 1.50 release
```
Comment (willarmiros) [inline integration-test-apps/auto-instrumentation/flask/create_flask_app.py:23]: Nit: Maybe remove the version number in this comment.

**Review by willarmiros — APPROVED** [review]
just some small stuff

File: integration-test-apps/auto-instrumentation/flask/application.py (hunk 1 above)
Comment (NathanielRN) [reply integration-test-apps/auto-instrumentation/flask/application.py:29]: Sounds good I will remove!

File: integration-test-apps/manual-instrumentation/flask/create_flask_app.py (hunk 2 above)
Comment (NathanielRN) [reply integration-test-apps/manual-instrumentation/flask/create_flask_app.py:20]: It is the logger used by the `flask` framework for its logs as I found out in [this SO post](https://stackoverflow.com/a/18379764/7460739). By "getting" that logger, we are modifying its log level (which will get very noisy when the Load Generator stress-tests the app).

File: integration-test-apps/auto-instrumentation/flask/create_flask_app.py (hunk 3 above)
Comment (NathanielRN) [reply integration-test-apps/auto-instrumentation/flask/create_flask_app.py:23]: Removed from both places!

**Pull request merged**